source/l4_testing/main.c

```diff
@@ -0,0 +1,142 @@
+#include "stm32l432xx.h"
+#include "can_parse.h"
+#include "common/psched/psched.h"
+#include "common/phal_L4/can/can.h"
+#include "common/phal_L4/gpio/gpio.h"
+
+#define BUTTON_1_Pin 5
+#define BUTTON_1_GPIO_Port GPIOB
+#define LED_GREEN_Pin 0
+#define LED_GREEN_GPIO_Port GPIOB
+#define LED_RED_Pin 1
+#define LED_RED_GPIO_Port GPIOB
+#define LED_BLUE_Pin 7
+#define LED_BLUE_GPIO_Port GPIOB
+
+GPIOInitConfig_t gpio_config[] = {
+  GPIO_INIT_CANRX_PA11,
+  GPIO_INIT_CANTX_PA12,
+  GPIO_INIT_OUTPUT(LED_GREEN_GPIO_Port, LED_GREEN_Pin, GPIO_OUTPUT_LOW_SPEED),
+  GPIO_INIT_OUTPUT(LED_RED_GPIO_Port, LED_RED_Pin, GPIO_OUTPUT_LOW_SPEED),
+  GPIO_INIT_OUTPUT(LED_BLUE_GPIO_Port, LED_BLUE_Pin, GPIO_OUTPUT_LOW_SPEED),
+  GPIO_INIT_INPUT(BUTTON_1_GPIO_Port, BUTTON_1_Pin, GPIO_INPUT_PULL_DOWN)
+};
+
+// Function Prototypes
+void canReceiveTest();
+void canSendTest();
+void Error_Handler();
+void SysTick_Handler();
+void canTxUpdate();
+
+q_handle_t q_tx_can;
+q_handle_t q_rx_can;
+
+int main (void)
```

Comment:
This looks like a good functional test of the PER HAL, scheduler, queues, and CAN generation!

Did testing this out work as expected?